Sw app shell

gauntface opened the pull request:
This is a version of the app shell that will load a single page web app from a SW , otherwise just use the static server render.

Commit:
First stab at serving up a SW which will override pages for the current origin

Commit:
Missing files?

addyosmani committed:
Patch package.json to work around estraverse issues

Commit:
Minor fixes to app controller and sw

Commit:
Fixing package merge conflict

Commit:
babelify not needed on sw

Commit:
It does need to be babelified

Commit:
Fixing up and logging requests going through sw

Commit:
Lots of logging

Commit:
Minor comments

Commit:
Adding a version that flips over app shell and takes over links in nav drawer

Commit:
Last commit broke gulp - boo

addyosmani commented:
Current issues while testing:

1. We need to add in focus styles for the menu and fix up the tab indexing. Right now we can't tab between menu items.
2. Do we want to have a class representing 'selected' state so we can highlight the current route in the menu?

(I can add these bits. Just wanted to check)

addyosmani commented:
Reviewed. Overall I'm happy with the approach. The one thing we'll need to figure out is a good flow for sharing router state (what's the current selected route?) with the UI for things like the selected menu state above.

addyosmani committed:
a11y: align with Material spec site

Can now tab into the menu and tab around items. This doesn’t yet
support proper arrow navigation with the keyboard but is better than
what we had before.

addyosmani committed:
a11y: hamburger -> add a border on focus/hover

addyosmani committed:
a11y: more improvements to focus styles

addyosmani committed:
sw.es6.js: drop redundant references

addyosmani committed:
UI: add simple cards for views

addyosmani commented:
Made a few improvements to a11y and added super simple cards for the views. Outstanding: route-based menu selection. I want to think some more about whether menu items should just be dynamically generated from the path-config and we use proper server-side logic to select the route vs doing client-side.

Otherwise, changes here LGTM for merging in.
